docs/source/developer/patterns.rst

```diff
@@ -174,3 +174,12 @@ Command Names
 Commands used in the application command registry should be formatted as
 follows: ``package-name:verb-noun``. They are typically grouped into a
 ``CommandIDs`` namespace in the extension that is not exported.
+
+Dialogs
```

Comment:
@JasonWeill, is something like this what you had in mind?

Reference: https://github.com/jupyterlab/jupyterlab/issues/16578#issuecomment-2231336738

Reply:
Yes, this is a good start! I also opened #16616 to break out the style guide into an issue of its own.

Reply:
Should I add something to this PR or is it good for now, @JasonWeill?